pyriemann/transferlearning.py

```diff
@@ -0,0 +1,144 @@
+
+import numpy as np
+from sklearn.model_selection import KFold
+from sklearn.base import BaseEstimator, TransformerMixin, ClassifierMixin
+from pyriemann.utils.mean import mean_riemann
+from pyriemann.utils.base import invsqrtm
+
+
+class TLSplitter():
+    def __init__(self, n_splits=5):
+        self.n_splits = n_splits
+
+    def split(self, X, y, meta):
+
+        # index of all source data points
+        idx_source = np.where(~meta['target'])[0]
+
+        # index of training-split for the target data points
+        kf_target = KFold(n_splits=self.n_splits).split(X[meta['target']])
+        for train_idx_target, test_idx_target in kf_target:
+            train_idx_target = list(
+                meta[meta['target']].index[train_idx_target])
+            train_idx = np.concatenate([idx_source, train_idx_target])
+            test_idx_target = list(meta[meta['target']].index[test_idx_target])
+            test_idx = test_idx_target
+            yield train_idx, test_idx
+
+    def get_n_splits(self, X, y, meta):
+        return self.n_splits
+
+
+class DCT(BaseEstimator, TransformerMixin):
+    '''
+    No transformation of the data points between the domains.
+    This is what we call the direct (DCT) method.
+    '''
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y, meta):
+        return self
+
+    def transform(self, X, y=None, meta=None):
+        return X
+
+    def fit_transform(self, X, y, meta):
+        return self.fit(X, y, meta).transform(X, y, meta)
+
+
+class RCT(BaseEstimator, TransformerMixin):
+    '''
+    Re-center (RCT) the data points from each domain to the Identity.
+    '''
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y, meta):
+        self._Minvsqrt = {}
+        for domain in np.unique(meta['domain']):
+            domain_idx = meta['domain'] == domain
+            M = mean_riemann(X[domain_idx])
```

Comment:
You could use `mean_covariance` with an argument `metric` set by default to 'riemann'.

Reply:
In fact, `preprocessing.Whitening.fit()` and `transform` could be re-used here.